### ISSUE TITLE
Refactor Rxjava to coroutines in FileSearching

### ISSUE THREAD

**Issue by gouri-panda** [kiwix/kiwix-android#2505]
**Is your feature request related to a problem? Please describe.**
Currently, we are using Rxjava for file searching. It would be great if we convert Rxjava to coroutines.

**Describe the solution you'd like**
Replace all the Rxjava methods with coroutines.
**Parent issue**
#2445

**Comment by kelson42** [comment]
@macgills Can you please validate the idea? Why should we do that?

**Comment by macgills** [comment]
I have approved it in the parent issue, coroutines are faster, easier to work with and more easily interruptible, stopping wasted cycles.

**Comment by s-ayush2903** [comment]
I would like to work on this issue

**Comment by gouri-panda** [comment]
@s-ayush2903 sorry, I am already start working on this.

**Comment by s-ayush2903** [comment]
Ah, fine then. But if we have any new issue on coroutines, I would like to work on it, I wanna learn implementing 'em efficiently

**Comment by gouri-panda** [comment]
@s-ayush2903 I don't think we have any new issue on coroutines. But if you like you can create new issues. Perhaps @macgills can help you with that.

**Comment by s-ayush2903** [comment]
A big thanks for your help! I'll create issues regarding the same after consulting the seniors and you, thanks again!

**Comment by kelson42** [comment]
@MohitMaliDeveloper @gouri-panda Does this ticket still make sense?

**Comment by gouri-panda** [comment]
> I have approved it in the parent issue, coroutines are faster, easier to work with and more easily interruptible, stopping wasted cycles.

@kelson42 We have discussed this before in the early of the ticket.

**Comment by kelson42** [comment]
How much work is it to implement?

**Comment by gouri-panda** [comment]
@kelson42 It's a little bit of work. I would suggest broadening up the issue instead of removing a part of Rxjava. We should remove all the Rxjava code from the project. It would be equivalent to the same as Room.

**Comment by kelson42** [comment]
@gouri-panda @MohitMaliFtechiz This issue has been postpone many times, but I really have for to implement it in the milestone 3.15
